### ISSUE TITLE
Password generator needs refactored . 

### ISSUE THREAD

**Issue by Pitoche** [quinone/password-manager#46]
The Password generator logic and UI need to be refactored 
I am currently having issues with the logic of character counting different types etc 
It is 40% working (being generous)

**Comment by quinone** [comment]
This is almost complete.
I still need to fix the form fields for the different types i.e password, passphrase or pin

**Comment by Pitoche** [comment]
Good job @quinone  @Rachie94 
Maybe we can incorporate this functionality directly on to the "New Item". That way, it can be used when adding a user and password 
just an idea... !!!!

**Comment by quinone** [comment]
Yeah its a good idea. I had started messing with it. If you see there is a file called password-widget.html.
It can be included on the page using {% include 'password-widget.html' %}
The issue I ran into was different 'POST' methods from the 2 different forms.
Any thoughts on a work around?
